### PR TITLE
SSCS 9874 Control Number

### PR DESCRIPTION
**Pull request by LTurkingtonHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-9874

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseDataTest.java [hunk 1]
```diff
@@ -367,6 +367,33 @@ public void givenACaseHasMultipleDocumentsOfSameTypeOnSameDayWithBundleAdditions
         assertEquals("Z11", sscsCaseData.getSscsDocument().get(7).getValue().getDocumentLink().getDocumentUrl());
     }
 
+    @Test
+    public void givenACaseHasScannedMultipleDocumentsOfSameDayWithControlNumber_thenSortByControlNumber() {
+        List<ScannedDocument> documents = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (LTurkingtonHMCTS) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseDataTest.java:372]: Using List.of() caused this error so we might need to keep it as arrayList

java.lang.UnsupportedOperationException
	at java.base/java.util.ImmutableCollections.uoe(ImmutableCollections.java:72)

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/AbstractDocumentDetails.java [hunk 2]
```diff
@@ -96,4 +98,9 @@ public Integer getSecondHalfOfBundleAddition() {
         }
         return NumberUtils.isCreatable(bundleAddition.substring(1)) ? Integer.parseInt(bundleAddition.substring(1)) : 0;
     }
+
+    public String getControlNumber() {
```
[the remark this answers is not in the document]
Comment (LTurkingtonHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/AbstractDocumentDetails.java:102]: Thanks, it's removed now

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/ScannedDocumentDetails.java [hunk 3]
```diff
@@ -49,4 +52,23 @@ public ScannedDocumentDetails(@JsonProperty("type") String type,
         this.originalSenderOtherPartyName = originalSenderOtherPartyName;
     }
 
+    @JsonIgnore
+    public LocalDateTime getScanDateTimeFormatted() {
+        try {
+            if (StringUtils.isEmpty(scannedDate)) {
```
[the remark this answers is not in the document]
Comment (LTurkingtonHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/ScannedDocumentDetails.java:58]: Updated

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/ScannedDocumentDetails.java [hunk 4]
```diff
@@ -49,4 +52,23 @@ public ScannedDocumentDetails(@JsonProperty("type") String type,
...
+            }
+            return LocalDateTime.of(LocalDate.parse(scannedDate), LocalTime.MIN);
+        } catch (DateTimeParseException e) {
+            return null;
+        }
+    }
+
+    @JsonIgnore
+    public Long getLongControlNumber() {
+        if (StringUtils.isEmpty(controlNumber)) {
```
[the remark this answers is not in the document]
Comment (LTurkingtonHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/ScannedDocumentDetails.java:69]: Thanks, updated also

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/AbstractDocumentDetails.java [hunk 5]
```diff
@@ -32,6 +32,7 @@ public class AbstractDocumentDetails {
     private String originalPartySender;
     private String originalSenderOtherPartyId;
     private String originalSenderOtherPartyName;
+    private String controlNumber;
```
[the remark this answers is not in the document]
Comment (LTurkingtonHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/AbstractDocumentDetails.java:35]: Updated :)

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseDataTest.java [hunk 6]
```diff
@@ -617,11 +617,11 @@ private DwpDocument buildDwpDocument(String documentUrl, DwpDocumentType documen
     private ScannedDocument buildScannedDocument(String documentUrl, DocumentType documentType, String date, String controlNumber) {
         String docType = documentType == null ? null : documentType.getValue();
         return ScannedDocument.builder().value(
-                ScannedDocumentDetails.builder().type(docType)
-                        .url(DocumentLink.builder().documentUrl(documentUrl).build())
-                        .scannedDate(date)
-                        .controlNumber(controlNumber)
-                        .build()).build();
+            ScannedDocumentDetails.builder().type(docType)
```
Comment (wmarshid) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseDataTest.java:620]: minor: formatting - can we new line the in-line builders as well? can show you what I mean on a call if you want.

**Review by wmarshid — APPROVED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Review by ikirsanov — APPROVED** [review]

**Pull request merged**